controllers/helmreleaseproxy/helmreleaseproxy_controller.go

```diff
@@ -177,6 +178,14 @@ func (r *HelmReleaseProxyReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, wrappedErr
 	}
 
+	if !conditions.IsTrue(cluster, clusterv1.ControlPlaneInitializedCondition) {
+		log.Info("Waiting for the control plane to be initialized")
+		conditions.MarkFalse(helmReleaseProxy, addonsv1alpha1.ClusterAvailableCondition, clusterv1.WaitingForControlPlaneAvailableReason, clusterv1.ConditionSeverityInfo, "")
+
+		// Requeue since the kubeconfig won't be available or useable until the API server is reachable.
+		return ctrl.Result{RequeueAfter: 15 * time.Second}, nil
```

Comment:
are we watching the `Cluster` objects? If so that would avoid us having to requeue every 15 seconds and simply requeue when the status of a Cluster changes

```suggestion
		return reconcile.Result{}, nil
```

Reply:
We aren't watching it on the HelmReleaseProxy but we are on the HelmChartProxy. I don't think we want to trigger a HelmReleaseProxy update whenever the Cluster changes, but I think it would make sense to watch only the status.

Reply:
You can use a predicate to watch that event specifically, see `ClusterControlPlaneInitialized` in the CAPI `predicates` package